src/components/KInput/KInput.vue

```diff
@@ -319,196 +238,148 @@ export default {
+  color: var(--kui-color-text-disabled, $kui-color-text-disabled);
+}
 
-      :deep(svg) {
-        height: $kInputMediumIconSize;
-        width: $kInputMediumIconSize;
-      }
-    }
+/* Component styles */
 
-    &.icon-start {
-      padding-left: calc($kInputMediumSizingX + var(--kui-space-40, $kui-space-40) + $kInputMediumIconSize) !important; // account for icon offset and width
-      ~ .input-icon {
-        left: $kInputMediumSizingX;
-      }
-    }
+.k-input-wrapper {
+  display: flex;
+  flex-direction: column;
+
+  // this doesn't work in Firefox
+  &:has(.k-input:disabled) {
+    cursor: not-allowed;
+  }
+  // workaround for Firefox
+  &.input-disabled {
+    cursor: not-allowed;
+  }
+
+  // error styles
+  &.input-error {
+    .k-input {
+      box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger);
 
-    &.icon-end {
-      padding-right: calc($kInputMediumSizingX + var(--kui-space-40, $kui-space-40) + $kInputMediumIconSize) !important; // account for icon offset and width
-      ~ .input-icon {
-        right: $kInputMediumSizingX;
+      // TODO: do we need a hover state for error?
+
+      &:focus {
+        box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger), var(--kui-shadow-focus, $kui-shadow-focus);
       }
     }
 
-    // input size small
-    $kInputSmallSizingY: var(--kui-space-40, $kui-space-40);
-    $kInputSmallSizingX: var(--kui-space-50, $kui-space-50);
-    $kInputSmallIconSize: var(--kui-icon-size-40, $kui-icon-size-40);
+    .help-text {
+      color: var(--kui-color-text-danger, $kui-color-text-danger);
+    }
+  }
 
-    &.k-input-small {
-      ~ .input-icon {
-        // (height of entire element (paddings + line height) - icon size) / 2
-        top: calc((($kInputSmallSizingY + $kInputSmallSizingY + $kInputLineHeight) - $kInputSmallIconSize) / 2);
+  .help-text {
+    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+    font-family: var(--kui-font-family-text, $kui-font-family-text);
+    font-size: var(--kui-font-size-20, $kui-font-size-20);
+    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
+    line-height: var(--kui-line-height-20, $kui-line-height-20);
+    margin: 0;
+    margin-top: var(--kui-space-40, $kui-space-40);
+  }
 
-        :deep(svg) {
-          height: $kInputSmallIconSize;
-          width: $kInputSmallIconSize;
+  // slots styles
+  .input-element-wrapper {
+    position: relative;
+
+    .before-content-wrapper, .after-content-wrapper {
+      position: absolute;
+      top: 50%;
+      transform: translateY(-50%);
+
+      :deep(>) {
+        &:is(.kui-icon) {
+          color: var(--kui-color-text-neutral, $kui-color-text-neutral) !important;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          height: $kInputIconSize;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          width: $kInputIconSize;
```

Comment:
question: why is the eslint disable comment needed here? Since these are SCSS variables, the stylelint plugin shouldn't be throwing any errors.